File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/model/ForecastDataBuilder.java
package org.rcgonzalezf.weather.openweather.model;

public class ForecastDataBuilder {
  private City city;
  private int count;

  public ForecastDataBuilder setCity(City city) {
    this.city = city;
    return this;
  }

  public ForecastDataBuilder setCount(int count) {
    this.count = count;
    return this;
  }

  public ForecastData createForecastData() {
    return new ForecastData(city, count);
  }
}

File: weather-library/src/main/java/org/rcgonzalezf/weather/openweather/model/ForecastDataBuilder.kt
package org.rcgonzalezf.weather.openweather.model

class ForecastDataBuilder {
    private lateinit var city: City
    private var count = 0
    fun setCity(city: City): ForecastDataBuilder {
        this.city = city
        return this
    }

    fun setCount(count: Int): ForecastDataBuilder {
        this.count = count
        return this
    }

    fun createForecastData(): ForecastData {
        return ForecastData(city, count)
    }
}
